fix typos

kaspar030 opened the pull request:
### Contribution description

This PR applies codespell fix suggestions throughout the codebase.

It is based on a rebased and squashed #12230.

### Testing procedure

<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references

<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

miri64 commented on drivers/enc28j60/enc28j60.c:
```diff
@@ -493,7 +493,11 @@ static void nd_isr(netdev_t *netdev)
         eir = cmd_rcr(dev, REG_EIR, -1);
     }
-    /* enable gobal interupt enable bit again */
+<<<<<<< HEAD
```
Please resolve this merge conflict ;-)

kaspar030 replied:
ups

miri64 commented on tests/pkg_tiny-asn1/main.c:
```diff
@@ -203,6 +203,6 @@ int main(void)
 
     /* Everything worked fine */
-    printf("Decoding finished succesfully\n");
+    printf("Decoding finished successfully\n");
```
Apparently this also needs to be fixed in the test script :woman_facepalming:

kaspar030 replied:
fixed

fjmolinas commented:
@kaspar030 this one needs rebase.

kaspar030 commented:
@francois-berder closed #12230, so this is not waiting for other PR's anymore (this includes the changes from that PR.

I realized that an improvement to codespell got in here accidentally. I reverted the commit here and split it into #12570.

kaspar030 commented:
> @kaspar030 this one needs rebase.

done

fjmolinas commented:
```
	cpu/kinetis/periph/ics.c:106: PRES  ==> PRESS
	cpu/sam0_common/periph/wdt.c:149: WEN  ==> WE, WHEN
	cpu/sam0_common/periph/wdt.c:151: WEN  ==> WE, WHEN
	cpu/sam0_common/periph/wdt.c:156: WEN  ==> WE, WHEN
	cpu/sam0_common/periph/wdt.c:158: WEN  ==> WE, WHEN
	cpu/stm32_common/include/periph_cpu_common.h:232: ot  ==> to, of, or
	cpu/stm32_common/include/periph_cpu_common.h:232: ot  ==> to, of, or
	drivers/lpsxxx/lpsxxx.c:157: pres  ==> press
	drivers/lpsxxx/lpsxxx.c:195: pres  ==> press
	pkg/openthread/include/ot.h:169: OT  ==> TO, OF, OR
```

what do we do about these? I feel like they are way to specific to add to `IGNORE`. I would just live with them.

Why is travis complaining about `Running './dist/tools/cppcheck/check.sh' x`?

kaspar030 commented:
> Why is travis complaining about `Running './dist/tools/cppcheck/check.sh' x`?

I think travis fails because this PR has fixme commits. cppcheck is the last check, but the warnings are not causing the failure.

fjmolinas commented:
> I think travis fails because this PR has fixme commits. cppcheck is the last check, but the warnings are not causing the failure.

But `cppcheck` is pointing out errors not warnings, or Am I missing something?

kaspar030 commented:
> But `cppcheck` is pointing out errors not warnings, or Am I missing something?

yeah, totally right. I guess the errors were already in there, and now that the files got touched, cppcheck checks them. the native irq_cpu seems to be a false positive. I'm checking the others.

kaspar030 commented:
Ok, there are a lot of cppcheck errors in master. Fixing them is too much for a supposed-to-be-trivial PR. I suggest splitting the offending commits out of this PR.

fjmolinas commented:
> Ok, there are a lot of cppcheck errors in master. Fixing them is too much for a supposed-to-be-trivial PR. I suggest splitting the offending commits out of this PR.

Yep, go ahead. Maybe open an issue on the cpp stuff.

kaspar030 commented:
Ok, with a lot of git, grep and shell magic I managed to filter all commits that touch files that are flagged by cppcheck. Unfortunately that includes @francois-berder's commit. (I'll probably cut that and filter by file).
Now I'm hitting another issue: cppcheck shows errors even of files included by checked files. That errors on files not even touched by a PR...

fjmolinas commented:
> Now I'm hitting another issue: cppcheck shows errors even of files included by checked files. That errors on files not even touched by a PR...

Hmm, I think the best way to tackle this would be to fix all the cpp errors first then. I'll try to find time.

fjmolinas commented:
@kaspar030 opened #12764 to try and unblock this :)

fjmolinas commented:
@kaspar030 this one needs rebasing now.

gschorcht commented:
@kaspar030 Great job :+1:

It's not wondering that the changes in `cpu/esp8266` have conflicts with the master because the new ESP8266 port in PR #11108 was merged in last days. Just discard your changes in `cpu/esp8266`. I have double checked, I already fixed the typos for `cpu/esp8266` when I was providing PR #11108.

kaspar030 commented:
> @kaspar030 Great job +1

Also @francois-berder, half this PR is his work.



> It's not wondering that the changes in `cpu/esp8266` have conflicts with the master because the new ESP8266 port in PR #11108 was merged in last days. Just discard your changes in `cpu/esp8266`. I have double checked, I already fixed the typos for `cpu/esp8266` when I was providing PR #11108.

Ok, I just skipped those. Thanks!

gschorcht commented:
> Also @francois-berder, half this PR is his work.

Of course, great job @francois-berder :+1:

fjmolinas commented:
@kaspar030 had the build pass except for you  last fix up?

kaspar030 commented:
> @kaspar030 had the build pass except for you last fix up?

Mostly. There were some warnings: some typos left, "Fix Typos" was bothering the PR check, and the header guard check correctly complained about the file fixed in the last fixup.

fjmolinas commented:
when running localy I still get some errors unflagged by murdock.

<details><summary><b>./dist/tools/codespell/check.sh</b></summary>

```
CODING_CONVENTIONS.md:47: seperate  ==> separate
core/include/clist.h:18: circularly  ==> circular, circularly
core/lifo.c:46: tThe  ==> the
cpu/arm7_common/include/arm7_common.h:72: BORD  ==> BOARD, BORED, BORDER
cpu/fe310/include/cpucycle.h:31: hart  ==> heart, harm
cpu/kinetis/include/cpu_conf_kinetis.h:108: PTD  ==> PDF
cpu/kinetis/periph/ics.c:106: PRES  ==> PRESS
cpu/lpc1768/periph/uart.c:162: THRE  ==> THREE, THERE, THEIR, THE
cpu/native/syscalls.c:465: creat  ==> create
cpu/sam0_common/periph/wdt.c:140: WEN  ==> WE, WHEN
cpu/sam0_common/periph/wdt.c:142: WEN  ==> WE, WHEN
cpu/sam0_common/periph/wdt.c:147: WEN  ==> WE, WHEN
cpu/sam0_common/periph/wdt.c:149: WEN  ==> WE, WHEN
cpu/stm32_common/include/periph_cpu_common.h:232: ot  ==> to, of, or
cpu/stm32_common/include/periph_cpu_common.h:232: ot  ==> to, of, or
cpu/stm32_common/periph/uart.c:376: UE  ==> DUE
cpu/stm32_common/periph/uart.c:377: UE  ==> DUE
cpu/stm32_common/periph/uart.c:388: UE  ==> DUE
dist/tools/esptool/espreset.py:62: ser  ==> set
dist/tools/esptool/espreset.py:63: ser  ==> set
dist/tools/esptool/espreset.py:65: ser  ==> set
dist/tools/esptool/espreset.py:66: ser  ==> set
dist/tools/esptool/espreset.py:70: ser  ==> set
dist/tools/esptool/espreset.py:71: ser  ==> set
dist/tools/esptool/esptool.py:2604: files'  ==> file's
drivers/include/ina3221.h:56: measurment  ==> measurement
drivers/include/lpsxxx.h:129: pres  ==> press
drivers/include/lpsxxx.h:134: pres  ==> press
drivers/include/mpl3115a2.h:160: pres  ==> press
drivers/include/mpl3115a2.h:166: pres  ==> press
drivers/include/opt3001.h:31: ALS  ==> ALSO
drivers/include/pir.h:120: whan  ==> want, when
drivers/lpsxxx/lpsxxx.c:157: pres  ==> press
drivers/lpsxxx/lpsxxx.c:195: pres  ==> press
drivers/mpl3115a2/mpl3115a2.c:155: pres  ==> press
drivers/mpl3115a2/mpl3115a2.c:168: pres  ==> press
drivers/mpl3115a2/mpl3115a2.c:169: pres  ==> press
drivers/mpl3115a2/mpl3115a2.c:169: pres  ==> press
drivers/w5100/w5100.c:105: extint  ==> extinct, extant
drivers/w5100/w5100.c:126: extint  ==> extinct, extant
examples/openthread/main.c:18: ot  ==> to, of, or
pkg/openthread/contrib/netdev/openthread_netdev.c:32: ot  ==> to, of, or
pkg/openthread/contrib/openthread.c:23: ot  ==> to, of, or
pkg/openthread/contrib/platform_alarm.c:23: ot  ==> to, of, or
pkg/openthread/contrib/platform_functions_wrapper.c:13: OT  ==> TO, OF, OR
pkg/openthread/contrib/platform_functions_wrapper.c:13: OT  ==> TO, OF, OR
pkg/openthread/contrib/platform_functions_wrapper.c:27: ot  ==> to, of, or
pkg/openthread/contrib/platform_functions_wrapper.c:211: OT  ==> TO, OF, OR
pkg/openthread/contrib/platform_radio.c:32: ot  ==> to, of, or
pkg/openthread/contrib/platform_uart.c:27: ot  ==> to, of, or
pkg/openthread/include/ot.h:169: OT  ==> TO, OF, OR
release-notes.txt:581: build-in  ==> built-in
release-notes.txt:806: crashs  ==> crashes
release-notes.txt:2033: witth  ==> with
release-notes.txt:2125: unconsistent  ==> inconsistent
release-notes.txt:2326: witth  ==> with
release-notes.txt:2375: unconsistent  ==> inconsistent
release-notes.txt:2925: agressive  ==> aggressive
release-notes.txt:2982: Eratic  ==> Erratic
release-notes.txt:3413: temperatur  ==> temperature
release-notes.txt:3465: agressive  ==> aggressive
release-notes.txt:3532: Eratic  ==> Erratic
release-notes.txt:3765: agressive  ==> aggressive
release-notes.txt:3815: numberic  ==> numeric
release-notes.txt:3837: Eratic  ==> Erratic
release-notes.txt:4213: Sytem  ==> System
release-notes.txt:5061: registeres  ==> registers
sys/arduino/SPI.cpp:72: mut  ==> must, mutt, moot
sys/arduino/SPI.cpp:77: mut  ==> must, mutt, moot
sys/arduino/SPI.cpp:90: mut  ==> must, mutt, moot
sys/arduino/SPI.cpp:95: mut  ==> must, mutt, moot
sys/arduino/SPI.cpp:108: mut  ==> must, mutt, moot
sys/arduino/include/spiport.hpp:108: mut  ==> must, mutt, moot
sys/arduino/include/wireport.hpp:78: adress  ==> address
sys/auto_init/auto_init.c:64: ot  ==> to, of, or
sys/include/net/ipv4/hdr.h:169: fo  ==> of, for
sys/include/net/ipv4/hdr.h:171: fo  ==> of, for
sys/include/net/ipv4/hdr.h:174: fo  ==> of, for
sys/include/net/ipv4/hdr.h:175: fo  ==> of, for
sys/posix/include/fcntl.h:59: creat  ==> create
sys/suit/v4/handlers.c:236: priotity  ==> priority
tests/bench_timers/print_results.c:111: pres  ==> press
tests/bench_timers/print_results.h:57: pres  ==> press
tests/driver_ina3221/main.c:29: thist  ==> this
tests/driver_lpsxxx/main.c:40: pres  ==> press
tests/driver_lpsxxx/main.c:47: pres  ==> press
tests/driver_lpsxxx/main.c:54: pres  ==> press
tests/gnrc_tcp/README.md:28: immediatly  ==> immediately
tests/gnrc_tcp/README.md:28: dispite  ==> despite
tests/gnrc_tcp/tests/06-receive_data_closed_conn.py:63: entirly  ==> entirely
tests/pkg_cn-cbor/main.c:2: ther  ==> there, their, the, other
tests/pkg_cn-cbor/main.c:16: ther  ==> there, their, the, other
tests/pthread_cooperation/main.c:28: ths  ==> the, this
tests/pthread_cooperation/main.c:61: ths  ==> the, this
tests/pthread_cooperation/main.c:66: ths  ==> the, this
tests/thread_cooperation/main.c:64: ths  ==> the, this
tests/thread_cooperation/main.c:73: ths  ==> the, this
tests/thread_cooperation/main.c:78: ths  ==> the, this
tests/thread_cooperation/main.c:83: ths  ==> the, this
tests/thread_cooperation/main.c:85: ths  ==> the, this
tests/unittests/tests-hashes/tests-hashes-sha1.c:54: BA  ==> BY, BE
tests/unittests/tests-hashes/tests-hashes-sha1.c:55: BA  ==> BY, BE
tests/unittests/tests-hashes/tests-hashes-sha1.c:107: BA  ==> BY, BE
tests/unittests/tests-sock_util/tests-sock_util.c:52: doesnt  ==> doesn't, does not
tests/unittests/tests-sock_util/tests-sock_util.c:55: doesnt  ==> doesn't, does not
```
</details>

I would add `ot`, `pres` to the ignore list.

I think some of the erros should only be filtered out for the specific files (e.g: `tests/unittests/tests-hashes/tests-hashes-sha1.c:107: BA  ==> BY, BE`). But that is a functionality to add in another PR for me.

kaspar030 commented:
- rebased, resolved conflicts

@fjmolinas this is passing now, but stales quickly. Needs an ACK...

kaspar030 commented:
> when running localy I still get some errors unflagged by murdock.

Can we postpone fixing them? This PR stales quickly, and fixing more typos requires going through all false positives again.

kaspar030 commented:
Thanks @gschorcht @fjmolinas!
